src/Features/Core/Portable/GenerateMember/GenerateVariable/AbstractGenerateVariableService.State.cs

```diff
@@ -148,7 +148,7 @@ internal bool CanGenerateLocal()
             internal bool CanGenerateParameter()
             {
                 // !this.IsInMemberContext prevents us offering this fix for `x.goo` where `goo` does not exist
-                return ContainingMethod != null && !IsInMemberContext && !IsConstant;
+                return ContainingMethod is { IsImplicitlyDeclared: false } && !IsInMemberContext && !IsConstant;
```

Comment:
@CyrusNajmabadi The test is failing because the compiler is returning `false` for `IsImplicitlyDeclared`.

Reply:
For some reason, ContainingMethod isn't even SynthesizedSimpleProgramEntryPointSymbol

![image](https://user-images.githubusercontent.com/31348972/94681158-43226b80-0323-11eb-9085-93cd7ead640c.png)

I've no idea why is that. As a workaround, I can rely on the method name.